src/charm.py

```diff
@@ -199,6 +193,19 @@ def __init__(self, *args):
         self.framework.observe(self.on.list_backups_action, self._on_list_backups_action)
         self.framework.observe(self.on.restore_backup_action, self._on_restore_backup_action)
 
+    def generate_vault_scrape_configs(self):
+        """Generate the scrape configs for the COS agent."""
+        return [
+                {
+                    "scheme": "https",
+                    "tls_config": {
+                        "insecure_skip_verify": False,
```

Comment:
Do we need to explicitly set this? seems like it should be the default...

Reply:
The default value is not explicitly mentioned in [prometheus docs. ](https://prometheus.io/docs/prometheus/latest/configuration/configuration/)

At least not for latest, the same config option has a default of false in alerting and in tracing configs, so I assume it is the case here too, so I don't insist on keeping it, but I feel the urge to remove it either.

I'm happy to adapt if you have a strong opinion here.

Reply:
I don't have a strong opinion either, it just seemed weird so I thought I'd ask.